### DISCUSSION TITLE
not allowed hyphen in dot notation?

### DISCUSSION THREAD

**Discussion by zucoinshur** [open-policy-agent/community#82] [OPA and Rego] [answered]
Hello
I'm using rego to validate a terraform plan, but there's a module with a hyphen in the name like "foo-baa"
If I execute: `data.configuration.root_module.module_calls.foo-baa` I get this error:
`1 error occurred: 1:1: rego_unsafe_var_error: var service is unsafe`
If I change the name to `foo_baa` it works. I guess it's because the dot notation wouldn't allow a hyphen. 
Is there a way I could still pass the JSON parameter with the hyphen?

**Comment by anderseknert** [comment, marked as answer]
Hyphen isn't valid in Rego identifiers, but is in JSON. If you have attributes in YAML/JSON containing hyphens in their names, you can reference them like:

```
data.configuration.root_module.module_calls["foo-baa"]
```

**Reply by zucoinshur** [reply]
Thank you, that worked!
